libecl/src/ecl_grid.c

```diff
@@ -3865,43 +3894,60 @@ bool ecl_grid_cell_contains_xyz1( const ecl_grid_type * ecl_grid , int global_in
         return false;
     }
 
+    if (ecl_cell_get_twist(cell) > 0) {
+      fprintf(stderr, "** Warning: Point (%g,%g,%g) is in vicinity of twisted cell: (%d,%d,%d) - function:%s might be mistaken.\n", x,y,z,i,j,k, __func__);
+      return false;
+    }
+
     {
-      double sign = 1.0;
-      int plane_nr = 0;
       double signed_volume = ecl_cell_get_signed_volume( cell );
-      if (fabs(signed_volume) > min_volume) {
+      if (fabs( signed_volume) > min_volume) {
+        double sign = 1.0;
+        int plane_nr = 0;
         point_type * p0;
         point_type * p1;
         point_type * p2;
 
+
         if (signed_volume < 0)
           sign = -1;
         {
           while (true) {
+
+            /* The distance has the right sign to all planes, i.e. we are inside. */
+            if (plane_nr == 12)
+              return true;
+
             p0 = &cell->corner_list[ bounding_planes[plane_nr][0] ];
             p1 = &cell->corner_list[ bounding_planes[plane_nr][1] ];
             p2 = &cell->corner_list[ bounding_planes[plane_nr][2] ];
 
-            if (point_equal(p0, p1) || point_equal(p0,p2) || point_equal(p1,p2))
-              return false;
+            if (point_equal(p0, p1) || point_equal(p0,p2) || point_equal(p1,p2)) {
+              plane_nr++;
+              continue;
+            }
 
             if (sign * point3_plane_distance(p0 , p1 , p2 , &p ) < 0)
               return false;
 
             plane_nr++;
-            if (plane_nr == 6)
-              return true;
           }
         }
-      } else
-        return false;
+      }
+
+      return false;
```

Comment:
This `while`-loop could preferably be written as a `for` loop.  Something like this?
```c
for (int plane_nr = 0; plane_nr < 12; plane_nr++) {
  p0 = &cell->corner_list[ bounding_planes[plane_nr][0] ];
  p1 = &cell->corner_list[ bounding_planes[plane_nr][1] ];
  p2 = &cell->corner_list[ bounding_planes[plane_nr][2] ];

  if (point_equal(p0, p1) || point_equal(p0,p2) || point_equal(p1,p2))
    continue;

  if (sign * point3_plane_distance(p0 , p1 , p2 , &p ) < 0)
    return false;
}
return true;
```